python/paddle/tensor/manipulation.py

```diff
@@ -37,6 +37,43 @@
 __all__ = []
 
 
+@dygraph_only
+def fill_diagonal_tensor_(x, src_tensor, dim1=0, dim2=1, offset=0):
```

Comment:
It seems that we can have another API named `fill_diagonal_tensor` (non-inplace version). This API can be used in both static and dygraph mode.

Reply:
done